src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1265,106 +1280,133 @@ protected override void ProcessRecord()
                             WriteVerbose(linkVerboseMsg);
                         }
 
-                        using (HttpRequestMessage request = GetRequest(uri, stripAuthorization:false))
+                        int retriesLeft = RetryCount;
```

Comment:
I believe it would be better to have `RetryCount` and `RetryIntervalSec` set on `WebSession` in `PrepareSession()` and then have this area of code retrieve the values from `WebSession`. It would mean extending `WebRequestSession` to include new Properties, but I believe that is the correct direction for these settings.

Reply:
Fixed.